[20.01] Release notes for 20.01

mvdbeek opened the pull request:
Please review and add anything I have forgotten.

mvdbeek committed:
Update scripts/bootstrap_history.py for python 3 / pygithub

mvdbeek committed:
Release notes for 20.01

bgruening commented on doc/source/releases/20.01_announce.rst:
```diff
@@ -3,8 +3,64 @@
- * Planned Release Date: 2020-01-27
+
+**Enhanced workflow functionality** 
+  Workflows can now now make use of optional datasets and optional parameters,
```
s/now//

bgruening commented on doc/source/releases/20.01_announce.rst:
```diff
@@ -3,8 +3,64 @@
+  Creating Job metadata has been reworked so that the dataset transfer that
+  occurs after a Galaxy job can be initiated by the compute node that runs a
+  Galaxy job. Previously the transfer was done by Galaxy under many production
+  scenarios.  This reduces IO load on Galaxy instances and should allow running
```
remove "many production scenarios."?

hexylena commented on doc/source/releases/20.01_announce.rst:
```diff
@@ -3,8 +3,64 @@
+  scenarios.  This reduces IO load on Galaxy instances and should allow running
+  many more jobs in parallel. This work will also make jobs running on remote
+  compute resources much more efficient if the remote resources can write to
+  Galaxy'so object store. To benefit from this new metadata setting Galaxy needs
```
```suggestion
  Galaxy's object store. To benefit from this new metadata setting Galaxy needs
```

mvdbeek committed:
Update doc/source/releases/20.01_announce.rst

Co-Authored-By: Helena Rasche <hxr@hx42.org>

mvdbeek committed:
Edit to 20.01_announce.rst

bernt-matthias commented on doc/source/releases/20.01.rst:
```diff
@@ -0,0 +1,655 @@
+* python3 fixes for setups using drmaa and realuser setups
+  (thanks to `@bernt-matthias <https://github.com/bernt-matthias>`__).
+  `Pull Request 9246`_
+* fix bug introduced in https://github.com/galaxyproject/galaxy/pull/9246
```
I guess this can be merged with the previous one. 

Btw. How can I add suggestion like @nsoranzo likes to do often, i.e. that you just need to accept a diff?

nsoranzo replied:
See https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/reviewing-proposed-changes-in-a-pull-request

mvdbeek committed:
First pass at user announements

hexylena commented on doc/source/releases/20.01_announce_user.rst:
```diff
@@ -8,20 +8,37 @@ January 2020 Galaxy Release (v 20.01)
+  This menu can be reached from the User menu -> Workflow Invocations
+  and lists your recent Workflow executions, their status and links
+  to the Workflow Editor and the History with the results of the
+  Workflow run.
```
Could you add a screenshot of how the new view looks? Maybe showing some recent in-progress/complete/etc. ones?

mvdbeek replied:
The screenhots are linked below.

hexylena replied:
Oh, that wasn't there when I commented / I missed the second one. I must've been on an earlier commit or just blind somehow. That looks great!

hexylena commented on doc/source/releases/20.01_announce_user.rst:
```diff
@@ -8,20 +8,37 @@ January 2020 Galaxy Release (v 20.01)
+
+**Email-notification for completed jobs**
+  You can now select to be notified by email when a tool run has finished.
+  Thanks goes to `@astrovsky01 <https://github.com/astrovsky01>`__
```
A screenshot would be nice here as well

hexylena replied:
perfect, thanks!

hexylena commented on doc/source/releases/20.01_announce_user.rst:
```diff
@@ -8,20 +8,37 @@ January 2020 Galaxy Release (v 20.01)
+
+**Galay Markdown Pages and Workflow Reports as PDF**
+  An additional link has been added to Galaxy Markdown Pahes and Workflow reports
+  that exports your document as a standalone PDF.
```
Maybe a picture of the report as PDF? How does it look when it's converted, if there's a galaxy output embedded or so?

mvdbeek committed:
Add workflow report and email button screenshots

mvdbeek committed:
Tweak image size and add 20.01_annouce to index

mvdbeek committed:
Fix RST formatting

mvdbeek committed:
Fix missing newline after section

nsoranzo committed:
Create Python3 virtualenv by default

- Allow specifying Python to use to create virtualenv using GALAXY_PYTHON
  environment variable.
- Integrate Python support docs from https://galaxyproject.org/admin/python/
- Update virtualenv installation to use v16.7.9

nsoranzo committed:
Python3: decode the output of ``subprocess.check_output()``

mvdbeek committed:
Pahes typo

nsoranzo commented on doc/source/releases/20.01.rst:
```diff
@@ -0,0 +1,626 @@
+  `Pull Request 9032`_
+* Add a config option to allow customizing the brand title.
+  `Pull Request 9062`_
+* Various interative tool fixes and enhancments including 3 new interative tools.
```
```suggestion
* Various interactive tool fixes and enhancements including 3 new interactive tools.
```

nsoranzo commented on doc/source/releases/20.01.rst:
```diff
@@ -19,235 +19,198 @@ Enhancements
   (thanks to `@nsoranzo <https://github.com/nsoranzo>`__).
   `Pull Request 8690`_
-* LDAP search member of filter
+* Allow utilizing LDAP search memberof filter during authenticating users
```
```suggestion
* Allow utilizing LDAP search ``memberof`` filter when authenticating users
```

nsoranzo commented on doc/source/releases/20.01.rst:
```diff
@@ -19,235 +19,198 @@ Enhancements
+* Add separate configuration knob for masthead colors.
   `Pull Request 8904`_
-* Refactor galaxy_install and required dependencies into Galaxy's namespace.
+* Refactor ``galaxy_install11 and required dependencies into Galaxy's namespace.
```
```suggestion
* Refactor ``galaxy_install`` and required dependencies into Galaxy's namespace.
```

jmchilton commented:
I was going to go through the rest of the notes but I'll stop there given the mess I made 😅😆.

astrovsky01 committed:
Added conda python update

nsoranzo committed:
Add more details

mvdbeek committed:
Update with new 20.01 PRs

mvdbeek committed:
Update with latest PRs

mvdbeek committed:
Merge followup PR into original announcement

mvdbeek committed:
Mention that database/compiled_templates needs to be removed when upgrading python

mvdbeek committed:
Link to Python upgrade instructions

mvdbeek commented:
I think this is ready to go now. Would be great if someone could double check the last 6 commits where I backported the Python upgrade instructions from dev.

mvdbeek committed:
Mention that more distribution tools/converters need dependencies now

nsoranzo commented on .circleci/config.yml:
```diff
@@ -97,24 +97,20 @@ jobs:
-      - run: sh scripts/common_startup.sh
-      - run: wget -q https://github.com/jmchilton/galaxy-downloads/raw/master/db_gx_rev_0127.sqlite
-      - run: mv db_gx_rev_0127.sqlite database/universe.sqlite
-      - run: sh manage_db.sh -c ./config/galaxy.yml.sample upgrade
```
The idea of leaving this here is to test the latest migrations, as in `.travis.yml` for macOS for py27. Maybe we can move this to py35 for CircleCI or TravisCI to have the full test matrix?

mvdbeek replied:
Sounds good to me, I'll move this to `py35_first_startup ` in circleci

mvdbeek replied:
Done in 4066d1bd757732fa7893f62e08e565a0a76d7331

nsoranzo commented on doc/source/admin/python.md:
```diff
@@ -0,0 +1,54 @@
+        1. Make sure a supported version of Python is installed.
+        2. Verify that the Python interpreter you want to use is first in the
+           output of `which -a python3 python` (or `which -a python` if you are
+           using Galaxy <=20.01). If this is not the case:
```
It's a bit silly to have the instructions for >=20.05 first, but I guess otherwise it becomes a mess to merge forward.

mvdbeek replied:
Since I cherry-picked 765f59175c3b3bdb1d83a1d1cdf824f3d522b96d we can change the versions now

mvdbeek committed:
Restore database migration test

mvdbeek committed:
Adjust docs for when we started preferring python3

nsoranzo commented on doc/source/admin/python.md:
```diff
@@ -0,0 +1,54 @@
+        2. Verify that the Python interpreter you want to use is first in the
+           output of `which -a python3 python` (or `which -a python` if you are
+           using Galaxy <=19.09). If this is not the case:
+           - If you are using Galaxy >= 20.05, just execute:
```
```suggestion
           - If you are using Galaxy >= 20.01, just execute:
```

mvdbeek committed:
One more version change

Co-Authored-By: Nicola Soranzo <nicola.soranzo@gmail.com>

mvdbeek committed:
Update for new 20.01 PRs

nsoranzo commented on scripts/bootstrap_history.py:
```diff
@@ -18,12 +19,14 @@
     requests = None
 try:
-    from pygithub3 import Github
+    from github import Github
```
Oh, that's great, it was on my TODO list for so long!

nsoranzo committed:
Use `with` statement to open file

nsoranzo committed:
Drop pygithub3 dev requirement
